pydantic/main.py

```diff
@@ -1102,8 +1110,16 @@ class RootModel(BaseModel, typing.Generic[RootModelRootType]):
 
     root: RootModelRootType
 
-    def __init__(__pydantic_self__, root: RootModelRootType) -> None:  # type: ignore
+    def __init__(
+        __pydantic_self__, root: RootModelRootType | Literal[_RootModelNoValue] = _RootModelNoValue
+    ) -> None:  # type: ignore
         __tracebackhide__ = True
+        if root is _RootModelNoValue:
+            root_field = __pydantic_self__.model_fields['root']
+            root = root_field.get_default(call_default_factory=True)
+            if not (__pydantic_self__.model_config.get('validate_default', False) or root_field.validate_default):
+                _object_setattr(__pydantic_self__, '__dict__', {'root': root})
+                return
```

Comment:
I would return `None` here. up to you to accept:
```suggestion
                return None
```

Reply:
I think that in this case it's better to use the bare `return` as an indication that return value has no particular meaning as opposite to the example above:

```py
def odd_or_none(value: int) -> int | None:
    if value % 2 == 0:
        return None
    return value
```

In this example we explicitly return `None` as the result of the function is going to be used somewhere and `None` would mean something to that logic.

On the other hand, in `__init__` methods and any method that always returns `None` just by convention the bare `return` emphasis the fact that the result is not going to be used.

Reply:
Agree!